src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Documents/EditorDocumentManager.cs

```diff
@@ -2,14 +2,15 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Razor;
 
 namespace Microsoft.VisualStudio.Editor.Razor.Documents
 {
     internal abstract class EditorDocumentManager : IWorkspaceService
     {
-        public abstract EditorDocument GetOrCreateDocument(
+        public abstract Task<EditorDocument> GetOrCreateDocumentAsync(
```

Comment:
Hmm, I'm curious why this needs to be async now. I'd imagine this would require the UI thread to be called.